### PR TITLE
chore: added package publishing steps to release workflow

### PR DESCRIPTION
**Pull request by pehlicd**
## How Has This Been Tested?
- [ ] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [ ] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [ ] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -10,10 +10,5 @@
 <!-- If so, please provide an explanation why it is necessary. -->
 
 ## How Has This Been Tested?
-- [ ] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
```
Comment (applike-ss) [inline .github/PULL_REQUEST_TEMPLATE.md:13]: will this be added again based on our template in our github management repo?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -117,3 +122,36 @@ jobs:
...
         run: echo "::set-output name=LINUX_BUILD_HASH::$(sha256sum ssm-diff-ubuntu-latest-${{ github.ref_name }}.zip | awk '{print $1}')"
+
+  containerize:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
```
Comment (applike-ss) [inline .github/workflows/release.yml:133]: based on this, it looks like you don't need this? https://docs.github.com/en/actions/use-cases-and-examples/publishing-packages/publishing-docker-images#publishing-images-to-github-packages

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+WORKDIR /ssm-diff
+
+COPY . .
+
+RUN pip install -r requirements.txt && \
+  pip install pyinstaller && \
+  pyinstaller --clean -y --onefile ssm-diff
+
+
+FROM public.ecr.aws/docker/library/alpine:3.20
```
Comment (applike-ss) [inline Dockerfile]: Can we use distroless here?

**Review by applike-ss — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 2 above)
Comment (pehlicd) [reply .github/workflows/release.yml:133]: actually it is needed for building multi-arch image

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (pehlicd) [reply .github/PULL_REQUEST_TEMPLATE.md:13]: I don't think so since we will probably not have any examples for this repository

Comment (applike-ss) [reply .github/PULL_REQUEST_TEMPLATE.md:13]: I know, that was also not my question. The file is managed however in out github management repository, isn't it? So the next apply will wipe out these changes, unless you blacklist it for this repo somehow.

File: .github/workflows/release.yml (hunk 2 above)
Comment (applike-ss) [reply .github/workflows/release.yml:133]: ahh isee

**Review by applike-ss — APPROVED** [review]

**Pull request merged**